Add table titles (#227)
riscv-non-isa/riscv-elf-psabi-doc: riscv-elf.adoc

e_flags:: Describes the format of this ELF file.  These flags are used by the
linker to disallow linking ELF files with incompatible ABIs together.
+

> 

Reviewer: This seems like it'd cause issues with the table no longer being sufficiently indented? The `+` is to tie the table as a continuation of the nested list item.

this support is implicit.

e_flags:: Describes the format of this ELF file.  These flags are used by the
linker to disallow linking ELF files with incompatible ABIs together,

> <<e-flags-layout>> show the layout of e_flags, and detail flags are listed

Reviewer: ```suggestion
<<e-flags-layout>> shows the layout of e_flags, and flag details are listed
```

| RVC   | Float ABI  | RVE   | TSO   | *Reserved*  | *Non-standard extensions*
|===

+

> --

Reviewer: I'd just keep the old couple of +'s, the use of `--` doesn't save you anything any more (I believe you could ditch the 145/150 ones, but two +'s vs two `--`'s doesn't really make a difference)

Author: I would prefer keep current layout, that made `Reserved` paragraph belong `e_flags` rather than `EF_RISCV_TSO`

Current layout:
![Selection_058](https://user-images.githubusercontent.com/2723185/143907474-7c7c61d5-b5c6-4e3e-a690-8f6941a2ceaf.png)

And the layout of the old couple of +'s:
![Selection_059](https://user-images.githubusercontent.com/2723185/143907695-fcd49423-b12d-4911-b28a-b48226f9ce18.png)

Reviewer: Oh I see. The current layout is what I intended. Though I think you can drop the `+`s within the `--`? Possibly also the `+` before the `--`?

Author: > Possibly also the + before the --?

Dropping that will cause `EF_RISCV_*` become same indention level as `e_flags`.

Reviewer: Weird. I'm the same, the documentation only covers the simple cases...

  EF_RISCV_TSO (0x0010)::: This bit is set when the binary requires the RVTSO
  memory consistency model.

> --
> +
> --

Reviewer: ```suggestion

```
should work, otherwise only getting limited benefit from the use of an open block

Author: Seems like remove those 3 lines not work well, that make next paragraph combine to EF_RISCV_TSO like this:
![Selection_057](https://user-images.githubusercontent.com/2723185/143903675-c1c0a7a8-56b5-47f8-8c6d-da2605194ae0.png)

Reviewer: GitHub screwed up my suggestion... was meant to be a single blank line

Author: Oh, cool, single blank line work.

Until such a time that the *Reserved* bits (0x00ffffe0) are allocated by future
versions of this specification, they shall not be set by standard software.
Non-standard extensions are free to use bits 24-31 for any purpose. This many
conflict with other non-standard extensions.

> [[e-flags-layout]]

Reviewer: Actually, why'd this table even get moved at all? It should at least be above the reserved and non-standard extensions bits paragraph above, but having it as the first thing was useful so you could see the summary and then dive into the details below if desired, rather than having to scroll through all the details. Was this originally just due to the missing `+`s messing up the indentation?

Author: My bad, I didn't understand this comment https://github.com/riscv-non-isa/riscv-elf-psabi-doc/pull/227#discussion_r754201167 well so made this change :(

Author: Moved back.

bit to mark functions that do not follow the standard calling convention for
the ABI in use.
+
The defined processor-specific `st_other` flags are listed in <<rv-st-other>>.
+

[[rv-st-other]]

> .RISC-V specific value for `st_other` flags.

Reviewer: ```suggestion
.RISC-V-specific `st_other` flags
```

==== Calculation Symbols

<<var-reloc-calc>> provides details on the variables used in relocation
calculation:

[[var-reloc-calc]]

> .Variable used in relocation calculation

Reviewer: ```suggestion
.Variables used in relocation calculation
```

==== Field Symbols

<<var-reloc-field>> provides details on the variables used in relocation fields:

[[var-reloc-field]]

> .Variable used in relocation fields

Reviewer: ```suggestion
.Variables used in relocation fields
```
Though variable(s) seems like the wrong word, these aren't variables, they're just names

past the end of the TCB.

There are various thread local storage models for statically allocated or
dynamically allocated thread local storage. <<tls-model>> lists the
thread local storage models:

[[tls-model]]

> .TLS model

Reviewer: ```suggestion
.TLS models
```

==== Section Types

The defined processor-specific section types are listed in <<rv-section-type>>.

[[rv-section-type]]

> .RISC-V specific section types

Reviewer: ```suggestion
.RISC-V-specific section types
```

==== Special Sections

<<rv-section>> lists the special sections defined by this ABI.

[[rv-section]]

> .RISC-V specific sections

Reviewer: ```suggestion
.RISC-V-specific sections
```

=== Program Header Table

The defined processor-specific segment types are listed in <<rv-seg-type>>.

[[rv-seg-type]]

> .RISC-V specific segment types

Reviewer: ```suggestion
.RISC-V-specific segment types
```

=== Dynamic Section

The defined processor-specific dynamic array tags are listed in <<rv-dyn-tag>>.

[[rv-dyn-tag]]

> .RISC-V specific dynamic array tags

Reviewer: ```suggestion
.RISC-V-specific dynamic array tags
```
